storage/flux/table.gen.go.tmpl

```diff
@@ -742,49 +743,21 @@ func (t *{{.name}}GroupTable) advance() bool {
 		return true
 	}
 
-	// handle the group with aggregate case
-	var value {{.Type}}
-	// For group count, sum, min, and max, the timestamp here is always math.MaxInt64.
-	// their final result does not contain _time, so this timestamp value can be anything
-	// and it won't matter.
-	// For group first, we need to assign the initial value to math.MaxInt64 so
-	// we can find the row with the smallest timestamp.
-	// Do not worry about data with math.MaxInt64 as its real timestamp.
-	// In OSS we require a |> range() call in the query and a math.MaxInt64 timestamp
-	// cannot make it through.
-	var timestamp int64 = math.MaxInt64
-	if t.gc.Aggregate().Type == datatypes.AggregateTypeLast {
-		timestamp = math.MinInt64
+	groupBy, err := determine{{.Name}}AggregateMethod(t.gc.Aggregate().Type)
+	if err != nil {
+		panic(err)
```

Comment:
I don't think that I should panic here, but I'm not sure how to handle this case.

Reply:
I agree. It seems to me that if it's not possible to guarantee that only supported aggregate types will be passed down here then ` advance() bool` needs to be refactored to ` advance() (bool, error)`.

It looks above that `t.do`, which takes the `advance` function returns an error, so perhaps this would be a relatively small refactor?